Allow custom API address

lawliet89 opened the pull request:
Due to a change in [Vault](https://github.com/hashicorp/vault/pull/4272), Vault will refuse to start up if `cluster_addr == api_addr`.

This PR incorporates #61 and furthers add a flag for the user to specify their own `api_addr` if they intend to use HA client redirection and they are running Vault behind a load balancer.

The documentation on [Vault's HA](https://www.vaultproject.io/docs/concepts/ha.html) page indicates that `cluster_addr` and `api_addr` should be top level keys.

I have tested this by bootstrapping my own Vault cluster successfully.

lawliet89 commented on modules/run-vault/run-vault:
```diff
@@ -148,11 +155,12 @@ $consul_storage_type "consul" {
+
+# HA settings
+cluster_addr  = "https://$instance_ip_address:$cluster_port"
+api_addr      = "$api_addr"
```
No. The default `api_addr` is `"https://${instance_ip_address}:${port}"`. Which is the port used for general Vault API access. This is different from `$cluster_port`.

lawliet89 commented on modules/run-vault/run-vault:
```diff
@@ -108,16 +109,22 @@ function generate_vault_config {
   instance_ip_address=$(get_instance_ip_address)
 
+
+  if [[ -z "$api_addr" ]]; then
```
Done.

lawliet89 commented:
`TestVaultClusterS3BackendAmazonLinuxAmi` apparently failed but I cannot find where the failure might have come from. 
[test.log](https://github.com/hashicorp/terraform-aws-vault/files/2015773/test.log)

brikis98 commented:
The error that caused the test failure:

> ssh.go:34: Process exited with status 1

This was probably after trying to unseal one of the hosts:

> TestVaultClusterS3BackendAmazonLinuxAmi 2018-05-18T14:29:34+08:00 vault_helpers.go:500: Unsealing Vault on host 13.210.183.243

Unfortunately, we're not getting much useful stdout/stderr from SSH.

Does this test case fail consistently?

jefferai commented:
It's super confusing to figure out what you changed in the readme -- can you put the whitespacing changes in a separate PR?

lawliet89 commented:
I have rebased the PR to resolve conflicts.

@jefferai You can ask Github to turn off whitespace only diffs.

@brikis98 I have rerun the test and updated the dependencies and the test passes.

[test.log](https://github.com/hashicorp/terraform-aws-vault/files/2024724/test.log)

brikis98 commented:
https://github.com/hashicorp/terraform-aws-vault/releases/tag/v0.7.1
